language/move-prover/tests/sources/functional/verify_table.move

```diff
@@ -0,0 +1,150 @@
+// This file verifies the intrinsic Table implementation.
+module 0x42::VerifyTable {
+    use Extensions::Table::{Self, Table};
+    use Extensions::Table::{spec_get, spec_len, spec_contains};
+
+    // TODO: test precise aborts behavior of all table functions
+
+    fun add(): Table<u8, u64> {
+        let t = Table::new<u8, u64>();
+        Table::add(&mut t, 1, 2);
+        Table::add(&mut t, 2, 3);
+        Table::add(&mut t, 3, 4);
+        t
+    }
+    spec add {
+        ensures spec_contains(result, 1) && spec_contains(result, 2) && spec_contains(result, 3);
+        ensures spec_len(result) == 3;
+        ensures spec_get(result, 1) == 2;
+        ensures spec_get(result, 2) == 3;
+        ensures spec_get(result, 3) == 4;
+    }
+
+    fun add_fail(): Table<u8, u64> {
+        let t = Table::new<u8, u64>();
+        Table::add(&mut t, 1, 2);
+        Table::add(&mut t, 2, 3);
+        Table::add(&mut t, 3, 4);
+        t
+    }
+    spec add_fail {
+        ensures spec_get(result, 1) == 1;
+    }
+
+    fun add_fail_exists(k1: u8, k2: u8): Table<u8, u64> {
+        let t = Table::new<u8, u64>();
+        Table::add(&mut t, k1, 2);
+        Table::add(&mut t, k2, 3);
+        t
+    }
+    spec add_fail_exists {
+        aborts_if k1 == k2;
+    }
+
+    fun remove(): Table<u8, u64> {
+        let t = add();
+        Table::remove(&mut t, 2);
+        t
+    }
+    spec remove {
+        ensures spec_contains(result, 1) && spec_contains(result, 3);
```

Comment:
I am curious if `!spec_contains(result, 3)` will be provable? I can figure it by testing it after this PR lands.

Reply:
Yes it should be.